src/Core/Framework/Struct/Collection.php

```diff
@@ -234,6 +234,36 @@ public function getIterator(): \Traversable
         yield from $this->elements;
     }
 
+    /**
+     * @param array<mixed> $associativeData
+     *
+     * @return Collection<TElement>
+     */
+    public function addFromAssociative(array $associativeData): self
+    {
+        $expectedClass = $this->getExpectedClass();
+        if ($expectedClass === null) {
+            return $this;
+        }
+
+        $baseObject = (new \ReflectionClass($expectedClass))->newInstanceWithoutConstructor();
+        $hasNecessaryInterface = $baseObject instanceof AssignArrayInterface;
+
+        foreach (\array_filter($associativeData) as $value) {
+            if ($hasNecessaryInterface && \is_array($value)) {
+                $value = (clone $baseObject)->assignRecursive($value);
+            }
+
+            try {
+                $this->add($value);
```

Comment:
Use the key too, collections can be serialized with IDs as keys:
```suggestion
                $this->set($key, $value);
```

Reply:
We should not do that. As you can see in my example in the PR description, we don't have any keys (uuid).
E.g. the `EntityCollection` will do `$this->set($entity->getUniqueIdentifier(), $entity);` inside the `add` method and we would miss that